test/integration/targets/azure_rm_virtualmachine/tasks/main.yml

```diff
@@ -1 +1,3 @@
+- include: setup.yml
 - include: virtualmachine.yml
+- include: teardown.yml
```

Comment:
the teardown.yml empty, pls remove it.

Reply:
Better yet, I will implement it properly so the test leaves nothing behind?

Reply:
By implementing it (teardown) I found a bug, where create_default_storage_account() on 2nd and subsequent invocations of the module can possibly make another storage account if the configuration changes, and the first one is forgotten.  This is because create_default_storage_account() should not be using the stored default_storage_account variable, but it should be checking the tags for '_own_sa_' and using that, since it is persistent across runs of the module.  So I will be fixing that as well.